src/xml/ConfigParser.cpp

```diff
@@ -214,7 +214,10 @@ void ConfigParser::connectTags(const ConfigurationContext &context, std::vector<
 
   for (auto &subtag : SubTags) {
     std::string expectedName = (subtag->m_Prefix.length() ? subtag->m_Prefix + ":" : "") + subtag->m_Name;
-    const auto  tagPosition  = std::find_if(
+    PRECICE_CHECK(expectedName != "solver-interface",
+                  "This configuration contains the tag <solver-interface>, meaning it was written for a preCICE version 1 or 2. "
```

Comment:
I think that "preCICE version 1" is clear that it includes all of 1.x. The "a preCICE version 1 or 2" sounds strange to me.

```suggestion
                  "This configuration contains the tag <solver-interface>, meaning it was written for preCICE version 1 or 2. "
```

Reply:
Alternative phrasing could be "this configuration compatible with versions prior to version 3"

Reply:
Both are fine and clear.

Reply:
```suggestion
                  "This configuration contains the tag <solver-interface>, meaning it was created for a preCICE version prior to version 3. "
```